packages/components/src/Disclosure/Disclosure.test.tsx

```diff
@@ -32,3 +33,23 @@ it("renders a Disclosure that is opened if `defaultOpen` is set", () => {
   const detailsElement = queryByRole("group");
   expect(detailsElement.getAttribute("open")).not.toBeNull();
 });
+
+describe("When a custom title is provided", () => {
+  it("renders the custom title", () => {
+    render(
+      <Disclosure
+        title={
+          <>
+            <Icon name="archive" />
+            <span>Custom Title</span>
+          </>
+        }
+      >
+        <span>Content</span>
+      </Disclosure>,
+    );
+
+    expect(screen.getByText("Custom Title")).toBeInTheDocument();
+    expect(screen.getByTestId("archive")).toBeInTheDocument();
+  });
+});
```

Comment:
I think it might be a good idea to add a test to make sure the "standard"/"string" `title` is not getting rendered when the "custom title" is passed. This might be not so straightforward, since we don't seem to be using an "id", "role" or other identifiers for string titles, but I see 2 alternatives:
1. Using a `toHaveClass` accessor to make sure `customSummaryWrap` class is present on a `div` (although you'd need some identifier on that `div`. Perhaps, `role="presentation"`?)
2. To check that `h4` element is NOT in the document (since this is what would be rendered in case the string was passed). However, this one might be a slightly riskier option, since the implementation of that title can be changed (e.g. from `<h4>` to `<p>`). Though we have snapshot testing in place to catch that, so shouldn't be a big issue.

Reply:
Makes sense!